infra/feast-operator/internal/controller/services/services.go

```diff
@@ -645,8 +652,10 @@ func (feast *FeastServices) getServerConfigs(feastType FeastServiceType) *feastd
 			return appliedServices.OnlineStore.Server
 		}
 	case RegistryFeastType:
-		if feast.isLocalRegistry() {
-			return appliedServices.Registry.Local.Server
+		if feast.isLocalRegistry() && appliedServices.Registry != nil &&
+			appliedServices.Registry.Local != nil &&
+			appliedServices.Registry.Local.Server != nil {
+			return &appliedServices.Registry.Local.Server.ServerConfigs
```

Comment:
`feast.isLocalRegistry()` already handles those checks.
```suggestion
		if feast.isLocalRegistry() {
			return &appliedServices.Registry.Local.Server.ServerConfigs
```